test-tools/aot-analyzer/src/main.cc

```diff
@@ -0,0 +1,558 @@
+    if (!sorted_func_ptrs) {
+        return NULL;
+    }
+
+    for (i = 0; i < module->func_count; i++) {
+        sorted_func_ptrs[i].idx = i;
+        sorted_func_ptrs[i].ptr = module->func_ptrs[i];
+    }
+
+    qsort(sorted_func_ptrs, module->func_count, sizeof(struct func_info),
+          compare_func_ptrs);
+    return sorted_func_ptrs;
+}
+
+static ObjdumpOptions s_objdump_options;
+
+static std::vector<const char *> s_infiles;
+
+static void
+ParseOptions(int argc, char **argv)
+{
+    OptionParser parser("aot-analyzer", s_description);
+
+    parser.AddOption('i', "info", "Print basic information about AoT",
+                     []() { s_objdump_options.info = true; });
+    parser.AddOption('t', "text-size", "Print text size",
+                     []() { s_objdump_options.text_size = true; });
+    parser.AddOption('x', "details", "Show AoT details",
+                     []() { s_objdump_options.details = true; });
+    parser.AddOption('c', "compare",
+                     "Show the differences between AoT and WASM",
+                     []() { s_objdump_options.compare = true; });
+    parser.AddArgument(
+        "filename", OptionParser::ArgumentCount::OneOrMore,
+        [](const char *argument) { s_infiles.push_back(argument); });
+    parser.Parse(argc, argv);
+}
+
+void
+InitStdio()
+{
+#if COMPILER_IS_MSVC
+    int result = _setmode(_fileno(stdout), _O_BINARY);
+    if (result == -1) {
+        perror("Cannot set mode binary to stdout");
+    }
+    result = _setmode(_fileno(stderr), _O_BINARY);
+    if (result == -1) {
+        perror("Cannot set mode binary to stderr");
+    }
+#endif
+}
+
+void
+DumpInfo(AoTFile *aot)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    printf("AOT File Information: \n\n");
+    printf("Binary type: %s \n",
+           aot->GetBinTypeName(target_info.bin_type).c_str());
+    printf("ABI type: %d \n", target_info.abi_type);
+    printf("Exectuion type: %s \n",
+           aot->GetExectuionTypeName(target_info.e_type).c_str());
+    printf("Exectuion machine: %s \n",
+           aot->GetExectuionMachineName(target_info.e_machine).c_str());
+    printf("Exectuion version: %d \n", target_info.e_version);
+    printf("Exectuion flags: %d \n", target_info.e_flags);
+    printf("Feature flags: %ld \n", target_info.feature_flags);
+    printf("Reserved: %ld \n", target_info.reserved);
+    printf("Arch: %s \n", target_info.arch);
+}
+
+void
+DumpTextSize(AoTFile *aot)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    printf("%s:       file format <%s> \n\n", aot->GetFileName(),
+           aot->GetBinTypeName(target_info.bin_type).c_str());
+    printf("Text size: \n");
+
+    const uint32_t literal_size =
+        ((AOTModule *)(aot->GetModule()))->literal_size;
+    const uint32_t code_size = ((AOTModule *)(aot->GetModule()))->code_size;
+    printf("   literal size= %d Bytes \n", literal_size);
```

Comment:
Is it better to use `%u` to dump uint32 values? And many places in this file.

Reply:
done